docs/elements/hx-search-control/index.html

```diff
@@ -0,0 +1,144 @@
+---
+title: <hx-search-control>
+minver: 0.18.0
+controlElement: <input type="search">
+also:
+  components/search: Search
+  elements/hx-search: <hx-search>
+  elements/hx-search-assistance: <hx-search-assistance>
+---
+{% extends 'element.njk' %}
+{% import '_utils.njk' as utils %}
+{% block page_header %}
+  <p>
+    The <code>{{page.title}}</code> block element is a container that augments
+    native browser validation and styles for an
+    <code>&lt;input type="search"&gt;</code> element.  Used to build
+    a <a href="components/search">Search</a> component.
+  </p>
+{% endblock %}
+
+{% block content %}
+  <section>
+    <dl class="metadata hxList">
+      <div>
+        <dt>Permitted Parents</dt>
+        <dd>any element that accepts flow content</dd>
+      </div>
+      <div>
+        <dt>Search Component API</dt>
```

Comment:
@CITguy, I'm proposing a way to document the `Search` component API.  Also, we'll have to update the other basic form controls with a facade.


<img width="800" alt="Screen Shot 2019-10-16 at 12 12 38 PM" src="https://user-images.githubusercontent.com/10120600/66944298-856b7380-f012-11e9-9c7e-fa41e38a8037.png">

Reply:
Introducing this change would be adding work that isn't on the roadmap for our v1.0.0 release, so I'd recommend keeping the existing format, for now.  However, we can revisit after the 1.0 release.

-----

Having said that, I'm curious to hear your reasoning behind the proposal, but I do have some concerns with the content.

* This seems like its a duplicate of the snippet for the demo on the Search component docs.
* Given that we're looking at the `<hx-search-control>` docs, including itself into the API seems redundant.
* The "indent" is hard to scan, given the uniform alignment of the numbers to the left.
    * Additionally, this is something that has to be applied manually, which adds mental overhead when writing documentation.
* None of the elements are actually optional (even the `<label>` is required, for a11y), so `(required)` is noisy and unnecessary.

Reply:
I'm am not tied to the documentation format.  Rather, I am more concerned with documenting how developers consume/implement the HelixUI Basic Form Controls API.

Namely:
1. As a developer, how do I consume HelixUI Basic Form Controls (components we've upgraded):
  * `Checkbox`
  * `Dropdown Select`
  *  `Radio`
  *  `Search`
  * `Textarea`
  * `Text Input`

2. As a developer, how do I implement HelixUI Basic Form Controls API:
  * What is the correct order of native elements and HelixUI custom elements to build a basic form control component?
  * What is required and what is optional?
  * If the component has a **_facade_**, where is it placed?
  * Can the component contain children or siblings?
 
Without properly capturing and documenting the above, we are placing a higher cognitive load on HelixUI consumers.